File: OOP-Java/Java Basics/p2/Car.java
package p2;

import p1.Vehicle;

class Car extends Vehicle {

    void print() {
        System.out.println(gameName);  //we can use it in sub class
    }

    public static void main(String[] args) {
        Vehicle g = new Vehicle();
       // g.gameName = "Basketball";      //Error: gameName has protected access in p1.Vehicle
        Car s = new Car();
        s.gameName = "Basketball";
    }
}

File: OOP-C++/Part3/Basics/1 Inheritance/Public Inheritance/Car.cpp
class Car : public Vehicle {

  public:
    int numGears;

    void print() {
        cout << "NumTyres : " << numTyres << endl; // Can access protected property as it is sub-class of Vehicle.
        cout << "Color : " << color << endl;       // Can access public property of Vehicle
        cout << "Num gears : " << numGears << endl;
        //cout<<"Max Speed : " << maxSpeed<<endl;		//maxSpeed is private member of Vehicle
    }
};

/*
   If we didn't specify access modifier while inheriting by default access modifier
   is private.
 */
